### PR TITLE
[fpv] Adding VC Formal support to the script and document

### PR DESCRIPTION
**Pull request by snpsverif**
Signed-off-by: Will Chen <haihui@synopsys.com>

### CODE REVIEW

File: hw/formal/README.md [hunk 1]
```diff
@@ -301,6 +302,31 @@ influence for "u_sha2.u_pad.shaf_ren".
 ...
 ```
 
+## How To Run VC Formal
```
Comment (sriyerg) [inline hw/formal/README.md]:
```suggestion
## Synopsys VC Formal
```

File: hw/formal/README.md [hunk 2]
```diff
@@ -301,6 +302,31 @@ influence for "u_sha2.u_pad.shaf_ren".
...
+
+If you have access to VC Formal from Synopsys, you can formally verify your
+assertions. For example, to run formal property verification (FPV) using
+VC Formal on module `gpio`, type:
+```
+  cd hw/formal
+  fpv gpio
```
Comment (sriyerg) [inline hw/formal/README.md]: This command is exactly the same for JasperGold as well? How does one distinguish between the two?

File: hw/formal/README.md [hunk 3]
```diff
@@ -252,8 +252,9 @@ can be used to add coverpoints to your design, where the cover
 property uses the same SVA syntax, operators, and system functions as the the
 assert properties.
 
-## How To Run JasperGold
+## How To FPV on OpenTitan
 
+### JapserGold
```
Comment (sriyerg) [inline hw/formal/README.md]:
```suggestion
### Cadence JapserGold
```

File: hw/formal/README.md [hunk 4]
```diff
@@ -252,8 +252,9 @@ can be used to add coverpoints to your design, where the cover
 property uses the same SVA syntax, operators, and system functions as the the
 assert properties.
 
-## How To Run JasperGold
+## How To FPV on OpenTitan
```
Comment (sriyerg) [inline hw/formal/README.md]:
```suggestion
## How to run FPV on OpenTitan
```

File: doc/ug/dv_methodology.md [hunk 5]
```diff
@@ -22,6 +22,7 @@ See discussion below on tracking progress.
 
 For professional tooling, the team has chosen [Synopsys' VCS](https://www.synopsys.com/verification/simulation/vcs.html) as the simulator of choice with respect to the tracking of verification completeness and [JasperGold](https://www.cadence.com/content/cadence-www/global/en_US/home/tools/system-design-and-verification/formal-and-static-verification/jasper-gold-verification-platform.html) for FPV.
 Wherever possible we attempt to remain tool-agnostic, but we must choose a simulator as our ground truth for our own confidence of signoff-level assurances.
+For FPV, [VC Formal](https://www.synopsys.com/verification/static-and-formal-verification/vc-formal.html) is also supported with the same flow and its own setup files, that could be used for those who have VC formal licenses.
```
Comment (sriyerg) [inline doc/ug/dv_methodology.md]: Move this after line 26, and change to:

```suggestion
Likewise, for FPV, [Synopsys VC Formal](https://www.synopsys.com/verification/static-and-formal-verification/vc-formal.html) is also supported within the same flow, and can be used by those with access to VC Formal licenses. 
```

File: hw/formal/fpv_vcf.tcl [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  create_clock tck_i -period 100
+  create_reset rst_ni -sense low
+  create_reset trst_ni -sense low
+} elseif {$env(FPV_TOP) == "spi_device"} {
+  create_clock clk_i -period 100
+  create_clock cio_sck_i -period 100
+  create_reset rst_ni -sense low
+  create_reset cio_csb_i -sense high 
+} elseif {$env(FPV_TOP) == "usb_fs_nb_pe"} {
+  create_clock clk_48mhz_i -period 100
```
Comment (sriyerg) [inline hw/formal/fpv_vcf.tcl]: As the name indicates, this is a 48MHz clock; period should be set accordingly?

File: hw/formal/README.md (hunk 2 above)
Comment (cindychip) [reply hw/formal/README.md]: We will change the `fpv` and `fpv_all` files to add a knob for vc_formal. With that PR, we will change the description here. Maybe it is good to add a TODO here.

File: hw/formal/fpv_vcf.tcl (hunk 6 above)
Comment (snpsverif) [reply hw/formal/fpv_vcf.tcl]: Hi Sri,

The exact frequency doesn’t matter for Formal verification. The usual convention is to first run all the clocks with same frequency to fix the design holes and keep the period simple numbers (to help with debug).
If you had 2 or more clocks then you would use this to set relative frequency, the next step is to maintain clock ratio right with whole numbers and then run (which not a lot of users do).

Let's know if you have more questions.

File: hw/formal/README.md (hunk 1 above)
Comment (cindychip) [reply hw/formal/README.md]: I think it should be `### Synopsys VC Formal`

**Comment by cindychip** [comment]
Hi team, just ping to see any other thoughts on this PR?

**Comment by sjgitty** [comment]
@cindychip let's discuss and try and confirm by EOD Monday. sg?

File: hw/formal/README.md (hunk 1 above)
Comment (cindychip) [reply hw/formal/README.md]: I will fix this in the coming PR.

**Pull request merged**